src/libaktualizr/uptane/uptane_test.cc

```diff
@@ -1113,12 +1113,18 @@ TEST(Uptane, DISABLED_IgnoreUnknownUpdate) {
   EXPECT_NO_THROW(sota_client->initialize());
   sota_client->AssembleManifest();
 
-  std::vector<Uptane::Target> targets;
-  EXPECT_TRUE(sota_client->uptaneIteration());
-  EXPECT_TRUE(sota_client->getNewTargets(&targets));
-
-  EXPECT_EQ(targets.size(), 1);
-  EXPECT_EQ(targets[0].filename(), "secondary_firmware.txt");
+  auto result = sota_client->fetchMeta();
+  EXPECT_EQ(result.status, result::UpdateStatus::kError);
+  EXPECT_STREQ(sota_client->getLastException().what(),
+               "The target had an ECU ID that did not match the client's configured ECU id.");
+  sota_client->last_exception = Uptane::Exception{"", ""};
+  result = sota_client->checkUpdates();
+  EXPECT_EQ(result.status, result::UpdateStatus::kError);
+  EXPECT_STREQ(sota_client->getLastException().what(),
+               "The target had an ECU ID that did not match the client's configured ECU id.");
+  std::vector<Uptane::Target> packages_to_install = UptaneTestCommon::makePackage("testecuserial", "testecuhwid");
+  auto report = sota_client->uptaneInstall(packages_to_install);
+  EXPECT_EQ(report.reports.size(), 0);
```

Comment:
Nice, this is great. It looks we don't actually reject unknown targets during installation, we just silently ignore them, which perhaps isn't ideal. But given how the code works now, this is what we need. Thanks!